src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -4175,6 +4176,12 @@ void StorageReplicatedMergeTree::removePartAndEnqueueFetch(const String & part_n
             part->was_removed_as_broken = true;
             part->makeCloneInDetached("broken", getInMemoryMetadataPtr(), /*disk_transaction*/ {});
             broken_part = part;
+
+            if (part->isStoredOnRemoteDiskWithZeroCopySupport() && part->storage.supportsReplication()
```

Comment:
Too broad, we should pass fetching only if key does not exist in object storage.

Reply:
yea. I will remove it.

Reply:
I will update.